lib/iris/tests/unit/cube/test_Cube.py

```diff
@@ -1457,255 +1348,127 @@ def test_lazy_data_masked__mask_set(self):
             data = ma.array([0, 1], dtype=dtype)
             data[0] = ma.masked
             cube = Cube(as_lazy_data(data))
-            self.assertEqual(cube.dtype, np.dtype('float'))
-            # Check that accessing the dtype does not trigger loading
-            # of the data.
-            self.assertTrue(cube.has_lazy_data())
-
-    def test_lazy_data_masked_with_dtype__mask_unset(self):
-        for dtype in self.dtypes:
-            data = ma.array([0, 1], dtype=dtype)
-            cube = Cube(as_lazy_data(data), dtype=dtype)
-            self.assertEqual(cube.dtype, dtype)
-            # Check that accessing the dtype does not trigger loading
-            # of the data.
-            self.assertTrue(cube.has_lazy_data())
-
-    def test_lazy_data_masked_with_dtype__mask_set(self):
-        for dtype in self.dtypes:
-            data = ma.array([0, 1], dtype=dtype)
-            data[0] = ma.masked
-            cube = Cube(as_lazy_data(data), dtype=dtype)
             self.assertEqual(cube.dtype, dtype)
             # Check that accessing the dtype does not trigger loading
             # of the data.
             self.assertTrue(cube.has_lazy_data())
 
-    def test_real_data_with_dtype_different(self):
-        for dtype in self.dtypes:
-            data = np.array([0, 1], dtype=dtype)
-            emsg = 'Cannot set realised dtype, no lazy data is available'
-            with self.assertRaisesRegexp(ValueError, emsg):
-                Cube(data, dtype=np.complex)
-
-    def test_lazy_data_masked_with_dtype__non_float_lazy_dtype(self):
-        for dtype in self.dtypes:
-            if dtype != np.dtype('float'):
-                data = ma.array([0, 1], dtype=dtype)
-                emsg = ("Cannot set realised dtype for lazy data "
-                        "with dtype\('{}'\)")
-                with self.assertRaisesRegexp(ValueError, emsg.format(dtype)):
-                    Cube(as_lazy_data(data), dtype=np.dtype('int16'))
-
-    def test_lazy_data_masked_with_dtype_non_integral(self):
-        for dtype in self.dtypes:
-            data = ma.array([0, 1], dtype=dtype)
-            data[0] = ma.masked
-            emsg = 'Can only cast lazy data to an integer or boolean dtype'
-            with self.assertRaisesRegexp(ValueError, emsg):
-                Cube(as_lazy_data(data), dtype=np.complex)
-
 
 class Test_data_dtype_fillvalue(tests.IrisTest):
-    def _sample_data(self, dtype='f4', masked=False, lazy=False):
+    def setUp(self):
+        self.fill_value = 1234
```

Comment:
Looks like you only use this in two other places, so one could argue you don't really need this?